src/Paramore.Brighter/OutboxProducerMediator.cs

```diff
@@ -1041,48 +1041,61 @@ private void OutstandingMessagesCheck(RequestContext? requestContext)
             }
         }
 
-        private bool Retry(Action action, RequestContext? requestContext)
+        private bool ExecuteWithResiliencePipeline(Action action, RequestContext? requestContext)
         {
-            var policy = _policyRegistry.Get<Policy>(CommandProcessor.RETRYPOLICY);
-            var result = policy.ExecuteAndCapture(action);
-            if (result.Outcome != OutcomeType.Successful)
+            var resiliencePipeline = _resiliencePipelineRegistry.GetPipeline(CommandProcessor.OutboxProducer);
+
+            try
             {
-                if (result.FinalException != null)
+                if (requestContext?.ResilienceContext != null)
                 {
-                    Log.ExceptionWhilstTryingToPublishMessage(s_logger, result.FinalException);
-                    CheckOutstandingMessages(requestContext);
+                    resiliencePipeline.Execute(_ => action, requestContext.ResilienceContext);
                 }
-
+                else
+                {
+                    resiliencePipeline.Execute(action);
+                }
+                
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Log.ExceptionWhilstTryingToPublishMessage(s_logger, ex);
+                CheckOutstandingMessages(requestContext);
                 return false;
             }
-
-            return true;
         }
 
-        private async Task<bool> RetryAsync(
+        private async Task<bool> ExecuteWithResiliencePipelineAsync(
             Func<CancellationToken, Task> send,
             RequestContext? requestContext,
             bool continueOnCapturedContext = true,
             CancellationToken cancellationToken = default)
         {
-            var policy = _policyRegistry.Get<AsyncPolicy>(CommandProcessor.RETRYPOLICYASYNC);
-            
-            var result = await policy
-                .ExecuteAndCaptureAsync(send, cancellationToken, continueOnCapturedContext)
-                .ConfigureAwait(continueOnCapturedContext);
+            var resiliencePipeline = _resiliencePipelineRegistry.GetPipeline(CommandProcessor.OutboxProducer);
 
-            if (result.Outcome != OutcomeType.Successful)
+            try
             {
-                if (result.FinalException != null)
+                if (requestContext?.ResilienceContext != null)
                 {
-                    Log.ExceptionWhilstTryingToPublishMessage(s_logger, result.FinalException);
-                    CheckOutstandingMessages(requestContext);
+                    await resiliencePipeline
+                        .ExecuteAsync(async context => await send(context.CancellationToken), requestContext.ResilienceContext)
+                        .ConfigureAwait(continueOnCapturedContext);
                 }
-
+                else
+                {
+                    await resiliencePipeline.ExecuteAsync(async _ => await send(cancellationToken), cancellationToken)
```

Comment:
You should pass through the cancellation token from the delegate arguments, rather than re-use `cancellationToken` in the closure, as the token might be different (e.g. combined with a token internal to Polly for handling timeouts).

For example:

```csharp
await resiliencePipeline.ExecuteAsync(async token => await send(token), cancellationToken)
```

Reply:
Thanks @martincostello. @lillo42 one to pick up.